rust/cmsis-pack/src/pdsc/device.rs

```diff
@@ -167,34 +167,18 @@ impl ProcessorBuilder {
         let name = self.name.clone();
 
         let map = (0..units)
-            .into_iter()
             .map(|unit| {
-                let default = Debug {
-                    ap: AccessPort::Index(0),
-                    dp: 0,
-                    address: None,
-                    svd: None,
-                    name: name.clone(),
-                    unit: None,
-                    default_reset_sequence: None,
-                };
-                let Debug {
-                    ap,
-                    dp,
-                    address,
-                    svd,
-                    name,
-                    default_reset_sequence,
-                    ..
-                } = debugs
-                    .iter()
-                    .find(|debug| {
-                        // If the <debug> element has a specific unit attribute we compare by that as well.
-                        // If not we just compare the name.
-                        let unit_condition = debug.unit.map(|u| u == unit).unwrap_or(true);
-                        debug.name == name && unit_condition
-                    })
-                    .unwrap_or_else(|| &default);
+                // The attributes we're interested in may be spread across multiple debug
+                // attributes defined in the family, subfamily, or device; and which may or may not
+                // be specific to a given Pname or Punit.
+                //
+                // We'll prioritize the first element with the attribute we're interested in, since
+                // family and subfamily debug elements are appended after device debug elements.
+                let debugs = debugs.iter().filter(|debug| {
```

Comment:
Can we call this `debug_iterator` (or what ever conveys that this is an iterator).
`debugs` suggest that we have a concrete list. It took me a couple second to understand it was fine to `clone()` as it’s actually only an itterator.